src/ActionModal/components/ActionModal.vue

```diff
@@ -746,6 +742,11 @@ export default {
         }
       },
       update(data) {
+        if (!data.overview) {
+          return {
+            totalRewards: 0
```

Comment:
shouldn't the API always return this, instead of adding conditionals to all our queries?

Reply:
what happens if the fetch from the API fails though?

Reply:
we should be using the error option, no?

https://vue-apollo.netlify.com/api/smart-query.html#options

Reply:
did you experiment with that? how should we use this? does this skip the update function?

Reply:
no i haven't tried it. from an architecture perspective, the API should always return the data we expect. if it doesn't there is likely an error with the API / server which we should catch the _vue-apollo_ way. 

the approach in this PR seems to be handling backend issues in the frontend which seems a bit funny to me. but i am not certain about this. and if this is a temporary fix for some problem we have now then we can certainly merge and address the underlying issues later.

https://learn.hasura.io/graphql/vue/queries/3-handle-errors

Reply:
I think there is a misunderstanding:
This doesn't handle issues with the backend but with the client losing its connection to the internet or similar issues. In this case, the backend can't serve the correct data.

Reply:
yeah, i think i get what you're saying. 

i'm suggesting two things:
1. let's use this vue-apollo `error` option for when there is no internet or the API is broken
2. let's make sure the API returns what the FE needs:

if `!data.overview` than the API is not working and `error` would catch it. shouldn't `data.overview` return `data.overview.totalRewards = 0`?
```
if (!data.overview) {
          return {
            totalRewards: 0
```

Reply:
ok, so you want to show an error "Couldn't get the data to show here" if there is any error? Or you want to show the technical error?

Reply:
this also doesn't solve the Sentry issues if the update function is still triggered and `data.overview` is undefined

Reply:
> this also doesn't solve the Sentry issues if the update function is still triggered and `data.overview` is undefined

But we need to know about these errors happening, to be able to fix them. Unless we are already successfully capturing the error in the API and this is just a duplicate.

Reply:
also should each component individually display an error? or a page displays an error?

Reply:
It depends. I think in the case of "floating" components like ActionModal it should be shown on the component itself and for "embedded" components it should appear at the top of the page, with small red letters (like we have them now), saying query X failed. IMO

Reply:
> But we need to know about these errors happening, to be able to fix them. Unless we are already successfully capturing the error in the API and this is just a duplicate.

we need to know every failing internet connection of a user?

Reply:
so do you think what you're doing in this PR is the right strategy for `update` queries?